frontend/src/components/inputs/AutoCompleteCourseSelect.js

```diff
@@ -46,11 +40,25 @@ class AutoCompleteCourseSelect extends Component {
   }
 
   onChange = (event, { newValue }) => {
+    clearTimeout(this.timer);
+
     this.setState({
       value: newValue
     });
+
+    this.timer = setTimeout(this.doSearch, WAIT_INTERVAL);
   };
 
+  doSearch = () => {
+    const requestUrl = apiConfig.coursesGeneral + '?search=' + this.state.value;
```

Comment:
```suggestion
    const requestUrl = apiConfig.coursesGeneral + '?search=' + encodeURIComponent(this.state.value);
```

Reply:
Cool, @OmarIthawi I made a note of this for when I get back into JavaScript programming again